packages/foam-workspace-manager/src/core.ts

```diff
@@ -0,0 +1,90 @@
+import { Graph, Edge } from 'graphlib'
+
+
+export interface BubbleLink {
+  from: string
+  to: string
+  text: string
+}
+
+
+export class Bubble {
+  public id: string
+  public title: string
+  public source: string
+  public path: string
+  public links: BubbleLink[]
+
+  constructor(id: string, title: string, links: BubbleLink[], path: string, source: string) {
+    this.id = id
+    this.title = title
+    this.source = source
+    this.path = path
+    this.links = links
+  }
+
+  public getForwardLinks(): BubbleLink[] {
+    return this.links.filter(link => link.from === this.id)
+  }
+
+  public getBacklinks(): BubbleLink[] {
+    return this.links.filter(link => link.from === this.id)
+  }
+}
+
+
+export class Foam {
```

Comment:
this is basically the interface to the graph of md notes. 
- it gives access to `Bubbles` and their connections
- it would be the object emanating events (I left that out for simplicity so far)
- it would manage refactoring operations (e.g. file rename)
- it is queried at higher levels (e.g. for backlinks)
- it can be validated for consistency (e.g. orphaned links, isolated nodes, ...)

Reply:
Personally I'd call this "NoteGraph" or something similar to make terminology obvious for newcomers (or maybe I'm just dull :)). Somehow the term "foam" in my head belongs to "branding layer" of the app, at the core it's still about notes, graphs etc. Especially in (maybe unlikely) case of a project name change from Foam to something else, things would be quite confusing for a while in the codebase.

Reply:
yeah I totally agree, this was me playing with ideas :) but I realize it can get confusing quickly

Reply:
I changed the name in the latest commit, so I will resolve this conversation